src/praisonai/tests/unit/sandbox/test_modal.py

```
@@ -0,0 +1,279 @@
+            result = await sandbox.execute_file("/nonexistent/file.py")
+            
+            assert result.status == SandboxStatus.FAILED
+            assert "File not found" in result.error
+    
+    @pytest.mark.asyncio
+    async def test_write_file_warning(self):
+        """Test write file shows warning for stateless functions."""
+        sandbox = ModalSandbox()
+        
+        with patch('praisonai.sandbox.modal.logger.warning') as mock_warning:
+            result = await sandbox.write_file("/tmp/test.py", "print('Hello')")
```

Ruff (0.15.10)

[error] Probable insecure usage of temporary file or directory: "/tmp/test.py"

(S108)